cmd/snap-bootstrap/README.md

```diff
@@ -12,10 +14,64 @@ _snap-bootstrap_ is the main executable that is run during the initramfs stage o
+
+Additionally, note that in all modes where encryption is enabled, we may open/unlock the TPM keys (or other security hardware in the case of custom ARM fde-hooks for example) in order to generally do encryption related things and regardless of result, we must _always_ re-seal and re-lock the TPM (or other security hardware) before exiting snap-bootstrap to protect the sealed-keys. This is implemented specifically with `secboot.LockSealedKeys`.
+
+### Install mode
+
+Install mode has the following steps:
+
+1. The first step of the initramfs-mounts command is always to measure the "epoch" of the secboot version that snap-bootstrap is compiled with to the TPM (if one exists). This is for maximum security and to prevent a newer epoch of secboot from being vulnerable to prior versions.
+1. The next step is to pick the first partition to mount as securely as possible, in the case of EFI systems, we will query an EFI variable to indicate the Partition UUID of the disk which the kernel was booted off of, and we use that Partition UUID to identify the partition which should be mounted as ubuntu-seed (since on grub amd64 systems, the kernel is initially booted by mounting the squashfs in grub and then booting the kernel.efi inside the mounted squashfs). If there is no such EFI variable, we fall back to just using the label instead to choose which partition to mount. Although we do have snap-bootstrap ordered to run after udev has fully settled via `After=systemd-udev-settle.service` in the unit file, sometimes we still don't have that Partition UUID device node available in /dev/ by the time we are executing, so we wait in a loop for the device node to appear before giving up.
+1. After having identified which partition is ubuntu-seed, we mount it at /run/mnt/ubuntu-seed.
+1. Next, we will load the "recovery system seed", which is the set of snaps associated with this recovery system, this includes the base snap, the kernel snap, the snapd snap and the gadget snap. These snaps are verified to match their assertions via hashing.
+1. Next we do another measurement to the TPM (if available) of the model assertion from the recovery system we loaded.
+1. After having verified that the recovery system seed snaps are valid and that the model assertion is correct, we will then mount these snaps at /run/mnt/base, /run/mnt/kernel, and /run/mnt/snapd (the gadget is not mounted at this time).
+1. Next, we create a tmpfs mount at /run/mnt/data, which will be the root filesystem we pivot_root into at the end of the initramfs. 
+1. Next, we will "configure" the target system root filesystem using the gadget snap itself, this will handle things like "early snap config" and cloud-init config, etc. that needs to be applied before we fully boot to userspace.
```

Comment:
```suggestion
1. Next, we will "configure" the target system root filesystem using the gadget snap itself, this will handle things like "early snap config" and cloud-init config, etc. that need to be applied before we fully boot to userspace.
```